pandas/core/frame.py

```
@@ -2091,10 +2091,23 @@ def to_feather(self, path) -> None:
     @Substitution(klass="DataFrame")
     @Appender(_shared_docs["to_markdown"])
     def to_markdown(
-        self, buf: Optional[IO[str]] = None, mode: Optional[str] = None, **kwargs
+        self,
+        buf: Optional[IO[str]] = None,
+        mode: Optional[str] = None,
+        index: Optional[bool] = None,
+        **kwargs,
     ) -> Optional[str]:
+        if index is not None and "showindex" in kwargs:
+            warnings.warn(
+                "Index and showindex are aliases. Setting them both "
+                "causes undefined behaviour and could be "
```

actually this should just raise an error